### PR TITLE
Arc - ignore synthetic methods when discovering observers and producers

### PR DESCRIPTION
**Pull request by mkouba**
- this should fix problems with lambdas on some old JDKs, see #3443
- also merge MethodUtils into the Methods class

### CODE REVIEW

**Review by manovotn — APPROVED** [review]
Looks good!

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java [hunk 1]
```diff
@@ -20,12 +20,19 @@
 import org.jboss.jandex.WildcardType;
 import org.jboss.logging.Logger;
 
+/**
+ * 
+ * @author Martin Kouba
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ */
```
Comment (gsmet) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: Didn't we decide to not add new `@author` tags?

Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: The deleted file (`MethodUtils`) had Michal as author and the functionality is now merged into this class, so the authors are just preserved.
Or is there a policy not to add `@author` at all to any files?

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: Arc is still developed as an "independent" project and I'd like to keep this tag ;-).

**Pull request merged**